pages/news/2020-XX-XX-new-in-2-3-hotcue-colors.html

```diff
@@ -0,0 +1,160 @@
+
+A small example:
+Say you want to set a cue point before the main vocal part, and you want it colored "blue".
+You can configure Mixxx to always assign "blue" to the hotcues number 5, and then just set hotcue number 5 at the desired position - done.
+
+If you do this, that also means that a hotcue of the same type (e.g. "before main vocals") does not only have the same color but is also assigned to the same hotcue pad.
+Hence, the possibility of getting confused during a gig is reduced even more than when just using colors as visual aid.
+
+{# TODO: Add GIF that shows color palette editor with indices and setting new hotcues with colors. #}
+
+## Bulk replace old hotcue colors
+
+At some point, you might feel like your current scheme for color code hotcues lacking and unintuitive.
+Maybe you simply don't like the colors anymore and would like to switch to a different palette.
+Or you realize that you find the colorful hotcue buttons actually distracting and want to get rid of them.
+If that happens, you may want to bulk replace the colors of existing hotcue in your library.
+
+We're not aware that any of our competitors in the DJ software market have such an option - Mixxx 2.3 has.
+You can find a dialog for replacing colors on the "Colors" page in the preferences window.
+
+It allows bulk replacing hotcue colors for hotcues in your library and can be used to
+- replace colors by hotcue number ("Set the color of all hotcues with number 5 to blue"),
+- replace colors by current color ("Set the color of all red hotcues to blue"),
+- combine the two conditions ("Set the color of all red hotcues with index 5 to blue"),
+- negate them ("Set the color of all hotcues that are red but don't have number 5 to blue")  or
+- just replace all hotcue colors in your library unconditionally ("Set the color of all hotcues to blue").
+
+{# TODO: Add screenshot or GIF of hotcue replace dialog. #}
+
+## Adding support for colored hotcues on controllers
+
+You're a controller mapping creator and wondered how to make use of hotcue colors on your controller?
+Then don't worry - we designed this feature with controller support in mind.
+
+When it comes to hotcue buttons, there are 4 types of controllers:
+1. Controllers without hotcue buttons
+2. Controllers with uncolored buttons
+3. Controllers with colored hotcue buttons that allow setting an arbitrary RGB color
+4. Controllers with colored hotcue buttons that only allow setting certain, predefined colors
+
+For controllers of types 1 and 2, nothing changes (obviously).
+
+For type 3, we made the RGB color of a hotcue accessible in controller scripts, so you can go ahead and implement support for it.
+If your controller mapping uses our JavaScript Components library, this is quite simple:
+You only need to implement the `HotcueButton::sendRGB(color)` method, that sends the 3-byte RGB color value to the controller (e.g. via a MIDI-SysEx message, depending on your hardware).
+
+What about type 4?
```

Comment:
Not sure if this section has to be as elaborate as it is. What about just summarizing what color mapper can do and put the example implementation on the wiki and link to it here?

Reply:
I agree. There is no need to put developer documentation in a blog post.

Reply:
I removed the source code.